dragonfly/roompolygon.py

```diff
@@ -0,0 +1,421 @@
+                model. If None, all boundary conditions will be Outdoors or Ground
+                depending on whether ceiling of the room is below 0 (the assumed
+                ground plane).
+            glazing_ratios: A list of numbers between 0 and 1 that match the number of
+                segments in the input Face3D geometry. These will be used to generate
+                window geometries for the walls in the resulting model. Note that these
+                values must be 0 for any Ground or Adiabatic walls.
+            glazing_parameters: A list of GlazingParameter objects that dictate how the
+                window geometries will be generated from the glazing_ratios. If None,
+                all window geometries will be generated by scaling the parent Face to
+                produce only one window per Face.
+            tolerance: The maximum difference between z values at which point vertices
+                are considered to be in the same horizontal plane. This is used to
+                check that all vertices of the input geometry lie in the same horizontal
+                floor plane. Default is None, which will not perform any check.
+        """
+        _Base.__init__(self, name)  # process the name
+
+        # process the geometry
+        assert isinstance(geometry, Face3D), \
+            'Expected ladybug_geometry Face3D. Got {}'.format(type(geometry))
+        self._geometry = geometry if geometry.normal.z >= 0 else geometry.flip()
+        self._segment_count = len(self.floor_segments)
+        self.floor_to_ceiling_height = floor_to_ceiling_height
+
+        # check that the geometry lies in the same plane.
+        if tolerance is not None:
+            z_vals = tuple(pt.z for pt in self._geometry.vertices)
+            assert max(z_vals) - min(z_vals) <= tolerance, 'Not all of RoomPolygon ' \
+                '"{}" vertices lie within the same horizontal plane.'.format(name)
+
+        # process the boundary conditions
+        if boundary_conditions is None:
+            bc = bcs.outdoors if self.ceiling_height > 0 else bcs.ground
+            self._boundary_conditions = [bc for i in range(len(self))]
+        else:
+            value = self._check_wall_assinged_object(
+                boundary_conditions, list, 'boundary_conditions')
+            for val in value:
+                assert isinstance(val, _BoundaryCondition), \
+                    'Expected BoundaryCondition. Got {}'.format(type(value))
+            self._boundary_conditions = value
+
+        # process the glazing ratio and glazing parameters
+        self.glazing_ratios = glazing_ratios
+        self.glazing_parameters = glazing_parameters
+
+        self._parent = None  # _parent will be set when RoomPolygon is added to a Story
+        self._properties = RoomPolygonProperties(self)  # properties for extensions
+
+    @property
+    def geometry(self):
```

Comment:
This is a design decision but I _feel_ geometry should return the full geometry not the floor.

Reply:
Ok but, in that case, what should we call this property? `floor_geometry`?

Reply:
`floor_geometry` sounds good to me.

Reply:
Ok. I will make it so.